Library/Homebrew/cmd/extract.rb

```diff
@@ -0,0 +1,53 @@
+#:  * `extract` <formula> <version> <tap>:
+#:    Looks through repository history to find the <version> of <formula> and
+#:    creates a copy in <tap>/Formula/<formula>@<version>.rb. If the tap is
+#:    not installed yet, attempts to install/clone the tap before continuing.
+#:
+#:    If the file at <tap>/Formula/<formula>@<version>.rb already exists,
+#:    it will not be overwritten unless `--force` is specified.
+#:
+
+require "utils/git"
+require "formula_versions"
+require "formulary"
+require "tap"
+
+module Homebrew
+  module_function
+
+  def extract
+    formula = Formulary.factory(ARGV.named[0])
```

Comment:
`ARGV.formulae.first` and needs handling for if a no arguments are provided.

Reply:
Replacing this line with `formula = ARGV.formulae.first` gives the result `Error: No available formula with the name "<version>"`.

Reply:
No worries, my mistake; didn't think about how you're using the other arguments. `ARGV.name.first` is sufficient (although you should check the value isn't `nil` first).